docs/examples/using-terraform-plugin.md

```diff
@@ -0,0 +1,265 @@
+    type: action
+    log: .
+    action:
+        secrets: ["CLIENT_ID", "TENANT_ID", "SUBSCRIPTION_ID", "CLIENT_SECRET"]
+        function: tfrun
+        input: 
+            action: "{{.action}}"
+            args-on-init: ["-backend-config=address=http://localhost:8001/terraform-azure-instance"]
+            execution-folder: terraform-examples/azure
+            variables:
+                state-name: terraform-azure-instance
+                subscription_id: "{{.secrets.SUBSCRIPTION_ID}}"
+                client_id: "{{.secrets.CLIENT_ID}}"
+                client_secret: "{{.secrets.CLIENT_SECRET}}"
+                tenant_id: "{{.secrets.TENANT_ID}}"
+    transform: |
+      {
+        "ip": .ip
+      }
+    transition: send_message
+```
+
+### Google Cloud Platform
+The only secrets required to run with Google is a service_account_key and a project_id.
+
+```yaml
+  - id: deploy_gcp
+    type: action
+    log: .
+    action:
+      secrets: ["SERVICE_ACCOUNT_KEY", "PROJECT_ID"]
+      function: tfrun
+      input:
+        execution-folder: terraform-examples/google
+        action: "{{.action}}"
+        args-on-init: ["-backend-config=address=http://localhost:8001/terraform-gcp-instance"]
+        variables:
+          state-name: terraform-gcp-instance
+          project_id: "{{.secrets.PROJECT_ID}}"
+          service_account_key: "{{.secrets.SERVICE_ACCOUNT_KEY}}"
+    transform: |
+      {
+          "ip": .ip
+      }
+    transition: send_message
+```
+
+### Amazon Web Services
+The only secrets required to run with Amazon is an access_key and a secret_key.
+
+```yaml
+  - id: deploy_amazon
+    type: action
+    log: .
+    action:
+      secrets: ["AMAZON_KEY", "AMAZON_SECRET"]
+      function: tfrun
+      input:
+        execution-folder: terraform-examples/amazon
+        action: "{{.action}}"
+        args-on-init: ["-backend-config=address=http://localhost:8001/terraform-amazon-instance"]
+        variables:
+          region: us-east-2
+          state-name: terraform-amazon-instance
+          amazon_key: "{{.secrets.AMAZON_KEY}}"
+          amazon_secret: "{{.secrets.AMAZON_SECRET}}"
+    transition: send_message
+    transform: |
+      {
+          "ip": .ip
+      }
+```
+
+**NOTE: you will find each terraform state has a transform attribute so I can easily get the IP to send via a Discord Message at the end**
+
+## Discord Message
+A simple action that sends a request to a Discord Webhook to post a message.
+
+```yaml
+  - id: send_message
+    type: action
+    action:
+      secrets: ["WEBHOOK_URL"]
+      function: discordmsg
+      input:
+        tts: false
+        url: "{{.secrets.WEBHOOK_URL}}"
+        message: "{{(\"The ip address of your  machine is \"+ .ip+\".\")}}"
+```
+
+## Full Example
+This is joining every state to deploy a machine to every cloud in one workflow. Also allows for an easy copy paste for you to try out and gives a better example of what the complete workflow looks like.
```

Comment:
Let's bring all of the states together to create a workflow that creates a virtual machine on Google Cloud Platform, Amazon Web Services, and Azure.